core/blockchain.go

```diff
@@ -2352,10 +2352,6 @@ func (bc *BlockChain) UpdateValidatorVotingPower(
 				utils.Logger().Debug().Err(err).Msg("issue with compute of apr")
 			}
 
-			if err != nil {
```

Comment:
I don't like this because now we are ignoring an error but its okay to ignore it because of the business logic carves out this way. 

The root problem is the APR return value semantics, let's fix that here as well since now we know that is correct & stable (the getting of the correct `*block.Header`) which was the original reason why the return error and value was so accompanding.

Reply:
right now it doesn't even get to the header, it is not able to get the `ReadValidatorSnapshotAtEpoch` `oneEpochAgo`. Which is a valid case when the validator is only created in the current epoch and does not exists in the previous epoch. The return value seems to be right below:
```
oneSnapshotAgo, err := bc.ReadValidatorSnapshotAtEpoch(
		oneEpochAgo,
		validatorNow.Address,
	)

	if err != nil {
		return &zero, err
	}
```